### ISSUE TITLE
feedback: Ablity to see history for edit code prompts

### ISSUE THREAD

**Issue by assyrus-favolo** [sourcegraph/cody#5200]
### Version

v1.30.3

### Areas for Improvement

- [X] UI/UX
- [ ] Onboarding
- [ ] Docs
- [ ] Chat
- [X] Commands
- [ ] Context
- [ ] Response Quality
- [ ] Other

### What needs to be improved? Please describe how this affects the user experience and include a screenshot.

I'd like to be able to pick up my old prompts, be it purely for archiving purposes or to save useful snippets that I might need to reuse later.  

### Describe the solution you'd like to see

Have a section where I can see the history of my prompts for code editing.  
Maybe with added context that could be the code generated, or simply the file path where it was generated, just to keep track of what the prompts are related to at least, but that's another issue.

### Describe any alternatives that could be considered

Merge that with the chat history, but I'm afraid that would make things confusing, unless the history tab became generic and included a filter on "conversation type", i.e. "chats" and "edit code prompts"

### Additional context

I've seen that an issue for this was already raised in #3427, though it was more an extra and it hasn't been addressed. I think it would still be a useful feature.

**Comment by jdorfman** [comment]
Thanks for reporting @assyrus-favolo I will make sure the product team sees this in the weekly community feedback report. =)

**Comment by rhettg** [comment]
Came here to leave the same request. I found myself today using Cody to edit a few different functions across different files in the same way. I sorta assumed I could just ⬆️ and do the same prompt, but sadly I had to copy and paste it to a temporary file so I could reuse it.

**Comment by jdorfman** [comment]
Great, thanks @rhettg I'm going to share this now since it is becoming a hot topic.

**Comment by assyrus-favolo** [comment]
Thank you, glad to see there's interest!

**Comment by ShalokShalom** [comment]
See also: https://github.com/sourcegraph/cody/issues/5318

**Comment by tosh** [comment]
I'd say this is the main feature missing for me in Cody right now that I run into 1-2 times per day.

For example sometimes editing errors out (not sure if network connection on my side or something from the llm that doesn't apply) and then the prompt is gone. Seems to happen more often for larger changes affecting 300+ lines.

In this case it would be great if the prompt would not be gone but possible to edit by me or re-run with another model, or change the selection and so on (similar to when I'm not happy with a diff and want to go back to editing the prompt).

So better error handling would already help but it would still be useful to have a history similar to chat history. Also for reviewing in hindsight what I have worked on and if there are certain patterns. It is good material to then do meta-prompts about.

Workaround: I now often re-write and then copy prompts into a text file when they errored out or with larger prompts I now copy them before I run them because in case it does not work I don't want to write it again (not great). The workaround works but is extra painful because often there is something else in the clipboard that I'm using at the moment.

In a nutshell: It breaks the flow.

**Comment by assyrus-favolo** [comment]
Hello, I see this has been closed, was this implemented? If so, is there a reference commit?

**Comment by ShalokShalom** [comment]
> Hello, I see this has been closed, was this implemented? If so, is there a reference commit?

They just closed 80+ commits without doing anything with them, they just try to get rid of perfectly fine issue reports.

Cody has become a fraud

**Comment by rhettg** [comment]
Doesn't appear that any behavior has changed around this feature.

**Comment by assyrus-favolo** [comment]
> > Hello, I see this has been closed, was this implemented? If so, is there a reference commit?
> 
> They just closed 80+ commits without doing anything with them, they just try to get rid of perfectly fine issue reports.
> 
> Cody has become a fraud

Ah, I see, that's weird. Not really the place for this kind of discussion, but I have to agree, it really rubs me wrong the way Sourcegraph is heading, this move feels very in the same spirit of the recent closing of the source for the code search product, opacity with the community.  
Hope to be wrong about that, because Cody is great in its own right
